data/data/agent/systemd/units/set-hostname.service

```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Agent-based installer hostname update service
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+ExecStart=/usr/local/bin/set-hostname.sh
+ExecStartPost=/bin/sleep 5
+Restart=on-failure
```

Comment:
Shouldn't it be:
```Type=oneshot```

Reply:
It looks like we should be setting hostnames only once and hence this systemd service should run only once
```
RemainAfterExit=true
Type=oneshot
```

Reply:
Yes, it should be oneshot.